src/main/java/bookstore/security/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package bookstore.security;
+
+import bookstore.repository.UserRepository;
+import jakarta.persistence.EntityNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class UserDetailsServiceImpl implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        return userRepository.findByEmail(email).orElseThrow(
+                () -> new EntityNotFoundException("User with such an email doesn't exist: " + email)
```

Comment:
```suggestion
                () -> new UsernameNotFoundException("User with such an email doesn't exist: " + email)
```
and remove it from the signature